File: library/src/main/java/me/proxer/library/api/chat/ReportChatMessageEndpoint.java
package me.proxer.library.api.chat;

import lombok.experimental.Accessors;
import me.proxer.library.api.Endpoint;
import me.proxer.library.api.ProxerCall;

/**
 * Endpoint for reporting a message to the moderators.
 *
 * @author Ruben Gees
 */
@Accessors(fluent = true)
public final class ReportChatMessageEndpoint implements Endpoint<Void> {

    private final InternalApi internalApi;

    private final String messageId;
    private final String message;

    ReportChatMessageEndpoint(final InternalApi internalApi, final String messageId, final String message) {
        this.internalApi = internalApi;
        this.messageId = messageId;
        this.message = message;
    }

    /**
     * {@inheritDoc}
     */
    @Override
    public ProxerCall<Void> build() {
        return internalApi.reportMessage(messageId, message);
    }
}

File: library/src/main/kotlin/me/proxer/library/api/chat/ReportChatMessageEndpoint.kt
package me.proxer.library.api.chat

import me.proxer.library.ProxerCall
import me.proxer.library.api.Endpoint

/**
 * Endpoint for reporting a message to the moderators.
 *
 * @author Ruben Gees
 */
class ReportChatMessageEndpoint internal constructor(
    private val internalApi: InternalApi,
    private val messageId: String,
    private val message: String
) : Endpoint<Unit?> {

    override fun build(): ProxerCall<Unit?> {
        return internalApi.reportMessage(messageId, message)
    }
}
